scripts/hid_configurator/configurator.py

```diff
@@ -1,151 +1,231 @@
+import os
 import hid
 import struct
 import time
-from enum import Enum
 
 import argparse
 import logging
+import collections
+
+from enum import IntEnum
 
 REPORT_ID = 4
-REPORT_SIZE = 8
+REPORT_SIZE = 29
+EVENT_DATA_LEN_MAX = REPORT_SIZE - 5
 
 NORDIC_VID = 0x1915
 MOUSE_PID = 0x52DE
 DONGLE_PID = 0x52DC
 
-POLL_INTERVAL = 0.5
-POLL_MAX = 5
+POLL_INTERVAL = 0.02
+POLL_RETRY_COUNT = 10
+
+
+GROUP_FIELD_POS = 6
+EVENT_GROUP_SETUP = 0x1
+EVENT_GROUP_DFU = 0x2
 
+MOD_FIELD_POS = 3
+SETUP_MODULE_SENSOR = 0x1
+SETUP_MODULE_LED = 0x1
 
-class ConfigEventID(Enum):
-    MOUSE_CPI = 0
-    DOWNSHIFT_RUN = 1
-    DOWNSHIFT_REST1 = 2
-    DOWNSHIFT_REST2 = 3
+OPT_FIELD_POS = 0
+SENSOR_OPT_CPI = 0x0
+SENSOR_OPT_DOWNSHIFT_RUN = 0x1
+SENSOR_OPT_DOWNSHIFT_REST1 = 0x2
+SENSOR_OPT_DOWNSHIFT_REST2 = 0x3
 
-class ConfigForwardStatus(Enum):
-	PENDING = 0
-	SUCCESS = 1
+class ConfigForwardStatus(IntEnum):
+	PENDING = 1
```

Comment:
swap places PENDING and SUCCESS to place it in order?

Reply:
Fixed